src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -370,18 +367,11 @@ public void CleanCache()
                                 break;
                             }
 
-                            if (!isAllProjectsNominated)
+                            if (!isAllProjectsNominated && retries++ >= DelayAutoRestoreRetries)
```

Comment:
Again I'm fine either way, personally i think debugging could be little tricky if not given due attention, otherwise it should be fine. 
@DoRonMotter